custom_components/localtuya/core/pytuya/__init__.py

```diff
@@ -902,12 +902,16 @@ def _status_update(msg, ack=False):
             listener = self.listener and self.listener()
             if listener is not None:
                 if cid:
-                    listener = listener.sub_devices.get(cid, listener)
-                    device = self.dps_cache.get(cid, {})
+                    # Don't pass sub-device's payload to the (fake)gateway!
+                    listener = listener.sub_devices.get(cid, None)
+                    status = self.dps_cache.get(cid, {})
                 else:
-                    device = self.dps_cache.get("parent", {})
+                    status = self.dps_cache.get("parent", {})
 
-                listener.status_updated(device)
+                if listener is not None:
+                    listener.status_updated(status)
+                else:
+                    self.debug(f"Payload for missing sub-device discarded: \"{decoded_message}\"")
```

Comment:
```suggestion
                if cid:
                    # Don't pass sub-device's payload to the (fake)gateway!
                    if not listener := listener.sub_devices.get(cid, None):
                        return self.debug(f"Payload for missing sub-device discarded: \"{decoded_message}\"")
                    status = self.dps_cache.get(cid, {})
                else:
                    status = self.dps_cache.get("parent", {})

                listener.status_updated(status)
```

Reply:
It didn't allow me to apply the change. Commited.